packages/backend/src/peripherals/coingecko/CoingeckoQueryService.ts

```diff
@@ -38,11 +40,15 @@ export class CoingeckoQueryService {
   }
 
   async queryPrices(
-    coinId: CoingeckoId,
+    assetId: AssetId,
     from: UnixTime,
     to: UnixTime,
     granularity: Granularity,
   ): Promise<Price[]> {
+    const coinId = tokenList.find((token) => token.id === assetId)?.coingeckoId
```

Comment:
This shouldn't work this way, because it introduces a hidden dependency on the token list. One way to resolve it would be to pass the token list in constructor, but I think that CoingeckoQueryService should only care about coingeckoIds and PriceUpdater should receive a list of all tokens and do the conversion there.